### ISSUE TITLE
The build fails in RealmConfiguration.swift in an xcode 7.2.1 project

### ISSUE THREAD

**Issue by atdash** [realm/realm-cocoa#3473]
## Goals
I want to use RealmSwift 0.99.0 in a new Xcode iOS Application project. I can create the project, init and install the RealmSwift using Cocoapods.. but the build immediately fails with errors in RealmConfiguration.swift

## Expected Results
I expect to be able to successfully build Xcode iOS application projects using RealmSwift 0.99.0. At this point I cannot create new projects that allow me to build
Question/workaround: Is there a way to provide a RealmSwift version override in the cocoapods? I tried "pod 'RealmSwift' '~> 0.98.8'" (and other combinations) with no successs

## Actual Results
After install RealmSwift using Pod Install, and reopening the .xcworkspace file, compilation completes, but the Build fails

## Steps to Reproduce
1. Create a new Xcode project
- iOS Application Single View Application
2. In the new application directory, install RealmSwift 0.99.0 using cocoapods
- pod init; edit pod to add "platform :ios, '8.0'', "use_frameworks!", and "pod 'RealmSwift'"; pod install
3. Open the project with the generated .xcworkspace file
- The files will index, but quickly fail with all errors in the RealmConfiguration.swift file
- There are two lines related to NSURL. Both display two matching error messages:

/Users/.../Pods/RealmSwift/RealmSwift/RealmConfiguration.swift:71:64: Expected expression in list of expressions

/Users/.../Pods/RealmSwift/RealmSwift/RealmConfiguration.swift:71:43: Cannot convert value of type 'NSURL' to expected argument type '@noescape (String) throws -> NSURL'

## Code Sample
Both lines fail at 'init' in  `.map(NSURL.init(fileURLWithPath:))` in RealmConfiguration.swift

A:
`                    objectTypes: [Object.Type]? = nil) {
            self.fileURL = path.map(NSURL.init(fileURLWithPath:))
            if inMemoryIdentifier != nil {
                self.inMemoryIdentifier = inMemoryIdentifier
            }
`
B:
`        public var fileURL: NSURL? {
            set {
                _inMemoryIdentifier = nil
                _path = newValue?.path
            }
            get {
                return _path.map(NSURL.init(fileURLWithPath:))
            }
        }
`
## Version of Realm and Tooling

ProductName:	Mac OS X
ProductVersion:	10.10.5
BuildVersion:	14F1713

/Applications/Xcode.app/Contents/Developer
Xcode 7.2.1
Build version 7C1002

/Users/Nicholas/.rvm/gems/ruby-2.2.1/bin/pod
0.39.0
Realm (0.99.0)
RealmSwift (0.99.0)

/bin/bash
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin14)

/usr/local/bin/carthage
0.15.2
(not in use here)

/usr/bin/git
git version 2.5.4 (Apple Git-61)

**Comment by Abreu0101** [comment]
Hi @atdash,

I also have the same problem, I'm only testing (first time checking Realm), but for the moment a downgrade the version (and right now I can build/run the app):
pod 'RealmSwift', '0.98.8'

**Comment by kishikawakatsumi** [comment]
@atdash @Abreu0101 

We can't reproduce this. One thing that's reportedly helped in these cases is clearing the CocoaPods cache and Xcode's Derived Data and trying to compile again. Could you please try doing this and reporting back again?

```
pod cache clean Realm
pod cache clean RealmSwift
pod deintegrate || rm -rf Pods
pod install --verbose
```

**Comment by BhavinBhadani** [comment]
@kishikawakatsumi same error. .. tried your solution ... still not working ...

**Comment by gelioson** [comment]
The same situation, both 0.99.0 and 0.99.1 versions produce  the same error in RealmConfiguration.swift
However 
`pod cache clean Realm
pod cache clean RealmSwift
pod deintegrate || rm -rf Pods
pod install --verbose`

and force
`pod 'RealmSwift', '0.98.8'`
in podfile solved the problem for me.

**Comment by gelioson** [comment]
One more thing.
The issue persists if EnableBitcode = No in pod target build settings
If I set EnableBitcode = Yes I can build and run the app successfully. But when I try to archive the app to create IPA the archiving process fails with the following error:
```
clang: error: unable to execute command: Segmentation fault: 11
clang: error: linker command failed due to signal (use -v to see invocation)
```

**Comment by jpsim** [comment]
We're glad to hear that clearing the CocoaPods cache solved this for you. In the future, CocoaPods cache issues should always be reported to the CocoaPods project rather than here.

For the problem you're having with bitcode, that shouldn't happen. Feel free to send us a project that demonstrates the issue and we'll analyze your project's configuration for you.

**Comment by vigbk** [comment]
I am getting the same issue as well with xcode 7.2.1 and realmswift 0.100.0. Attached a sample project.
[Test.zip](https://github.com/realm/realm-cocoa/files/249672/Test.zip)

**Comment by jpsim** [comment]
Alright, this was an issue on our end, the Swift syntax we were using didn't compile under Swift 2.1, it was a simple fix, included in 642403c.
